src/main/java/ru/testtask/service/WebSocketEventListener.java

```diff
@@ -0,0 +1,44 @@
+package ru.testtask.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.event.EventListener;
+import org.springframework.messaging.simp.SimpMessageSendingOperations;
+import org.springframework.messaging.simp.stomp.StompHeaderAccessor;
+import org.springframework.stereotype.Component;
+import org.springframework.web.socket.messaging.SessionConnectedEvent;
+import org.springframework.web.socket.messaging.SessionDisconnectEvent;
+import ru.testtask.model.ChatMessage;
+import ru.testtask.model.MessageType;
+
+import static java.lang.String.format;
+
+@Slf4j
+@Component
+public class WebSocketEventListener {
+
+    @Autowired
+    private SimpMessageSendingOperations messagingTemplate;
+
+    @EventListener
+    public void handleWebSocketConnectListener(SessionConnectedEvent event) {
+        log.debug("Received a new web socket connection");
+    }
+
+    @EventListener
+    public void handleWebSocketDisconnectListener(SessionDisconnectEvent event) {
+        StompHeaderAccessor headerAccessor = StompHeaderAccessor.wrap(event.getMessage());
+
+        String username = (String) headerAccessor.getSessionAttributes().get("username");
+        String roomId = (String) headerAccessor.getSessionAttributes().get("room_id");
+        if (username != null) {
+            log.debug("User Disconnected: " + username);
+
+            ChatMessage chatMessage = new ChatMessage();
```

Comment:
You send "leave" message in 2 cases: 1) user change the room, 2) user disconnected. It isn't good idea to implemetnt this functionality twice. Please create universal method to `ChatMessageService`:
`public void sendUserLeaveFromRoom(String username, String roomId)`